ambry-store/src/test/java/com.github.ambry.store/CuratedLogIndexState.java

```diff
@@ -168,11 +175,13 @@ void destroy() throws InterruptedException, IOException, StoreException {
    * @throws IOException
    * @throws StoreException
    */
-  FileSpan addPutEntries(int count, long size, long expiresAtMs) throws IOException, StoreException {
+  Pair<FileSpan, List<IndexEntry>> addPutEntries(int count, long size, long expiresAtMs)
+      throws IOException, StoreException {
     if (count <= 0) {
       throw new IllegalArgumentException("Number of put entries to add cannot be <= 0");
     }
-    ArrayList<IndexEntry> indexEntries = new ArrayList<>(count);
+    ArrayList<IndexEntry> newEntries = new ArrayList<>(count);
+    ArrayList<IndexEntry> indexEntries = new ArrayList<>(1);
```

Comment:
You don't need `newEntries` here. The old code works ok - just `addToIndex` in each iteration in the loop and remove the `addToIndex` that was outside.